### PR TITLE
Sessions: add/edit: validate gracePeriod values #8643

### LINKED ISSUE #8643: Sessions: add/edit: validate gracePeriod values
Noticed while working on #8641 - there is no current validation for `gracePeriod`. Though the UI presents a dropdown list, manipulating the DOM allows users to specify any integer value, including negative values. These are successfully persisted to the datastore.

### PR DESCRIPTION
**Pull request by shaharyar-shamshi**
Fixes #8643 
Validation of the grace period of the session while adding or editing the session

### CODE REVIEW

**Comment by shaharyar-shamshi** [comment]
ongoin logic added

**Comment by shaharyar-shamshi** [comment]
@whipermr5 sir ready for review

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 1]
```diff
@@ -320,6 +323,17 @@ public String getInvalidityInfoForEmail(String email) {
         return "";
     }
 
+    /**
+     * Checks if {@code gracePeriod} is not negative.
+     * @return An Explanation why the {@code gracePeriod} not acceptable.
```
Comment (whipermr5) [inline src/main/java/teammates/common/util/FieldValidator.java]: An ~~E~~**e**xplanation why the `gracePeriod` **is** not acceptable.

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 2]
```diff
@@ -153,6 +154,8 @@
             "The provided ${fieldName} is not acceptable to TEAMMATES as it cannot be empty.";
 
     // field-specific error messages
+    public static final String GRACE_PERIOD_NEGATIVE = " Grace period should never be negative i.e. "
+            + " it should always be positive or zero";
```
Comment (whipermr5) [inline src/main/java/teammates/common/util/FieldValidator.java]: This doesn't seem necessary.

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 3]
```diff
@@ -153,6 +154,8 @@
             "The provided ${fieldName} is not acceptable to TEAMMATES as it cannot be empty.";
 
     // field-specific error messages
+    public static final String GRACE_PERIOD_NEGATIVE = " Grace period should never be negative i.e. "
```
Comment (whipermr5) [inline src/main/java/teammates/common/util/FieldValidator.java]: Don't start with a space.

File: src/test/java/teammates/test/cases/datatransfer/FeedbackSessionAttributesTest.java [hunk 4]
```diff
@@ -138,4 +138,28 @@ public void testValidate() {
         return Arrays.asList(feedbackSessionNameError, courseIdError, creatorEmailError);
     }
 
+    @Test
+    public void testGracePeriodValidate() {
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/datatransfer/FeedbackSessionAttributesTest.java]: Combine this with `testValidate` instead. Simply modify the builder and add the new error to `buildExpectedErrorMessages`.

**Comment by shaharyar-shamshi** [comment]
@whipermr5 sir I have made all the suggested changes kindly review

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 5]
```diff
@@ -153,6 +154,7 @@
             "The provided ${fieldName} is not acceptable to TEAMMATES as it cannot be empty.";
 
     // field-specific error messages
+    public static final String GRACE_PERIOD_NEGATIVE = "Grace period should never be negative. ";
```
Comment (whipermr5) [inline src/main/java/teammates/common/util/FieldValidator.java]: Trailing space in the message

**Comment by shaharyar-shamshi** [comment]
@whipermr5  sir I made suggested changes kindly review it

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 6]
```diff
@@ -320,6 +322,17 @@ public String getInvalidityInfoForEmail(String email) {
         return "";
     }
 
+    /**
+     * Checks if {@code gracePeriod} is not negative.
+     * @return An explanation why the {@code gracePeriod} is not acceptable.
+     */
```
Comment (whipermr5) [inline src/main/java/teammates/common/util/FieldValidator.java]: For consistency with the other methods, add a line `Returns an empty string if the {@code gracePeriod} is acceptable.`

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 7]
```diff
@@ -153,6 +154,7 @@
             "The provided ${fieldName} is not acceptable to TEAMMATES as it cannot be empty.";
 
     // field-specific error messages
+    public static final String GRACE_PERIOD_NEGATIVE = "Grace period should never be negative.";
```
Comment (whipermr5) [inline src/main/java/teammates/common/util/FieldValidator.java]: never -> not

File: src/test/java/teammates/test/cases/datatransfer/FeedbackSessionAttributesTest.java [hunk 8]
```diff
@@ -134,8 +135,8 @@ public void testValidate() {
         String creatorEmailError = "The field 'email' is empty. An email address contains some text followed "
                 + "by one '@' sign followed by some more text. It cannot be longer than 254 characters, cannot be empty"
                 + " and cannot contain spaces.";
+        String gracePeriodError = "Grace period should never be negative.";
 
-        return Arrays.asList(feedbackSessionNameError, courseIdError, creatorEmailError);
+        return Arrays.asList(feedbackSessionNameError, courseIdError, creatorEmailError, gracePeriodError);
     }
-
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/datatransfer/FeedbackSessionAttributesTest.java]: Can revert this removal of a blank line as it's an unrelated change.

File: src/main/java/teammates/common/util/FieldValidator.java (hunk 7 above)
Comment (whipermr5) [reply src/main/java/teammates/common/util/FieldValidator.java]: Also, move this to after the time zone related messages. Can also add a hint: `The value must be one of the options in the grace period dropdown selector.` (actually, you might want to validate that too, though not as important)

**Comment by shaharyar-shamshi** [comment]
@whipermr5 Sir all suggested change made kindly review it

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 9]
```diff
@@ -181,6 +182,10 @@
             "The value must be one of the values from the time zone dropdown selector.";
     public static final String COURSE_TIME_ZONE_ERROR_MESSAGE =
             ERROR_INFO + " " + HINT_FOR_CORRECT_COURSE_TIME_ZONE;
+    public static final String HINT_FOR_CORRECT_GRACE_PERIOD =
+            "The value must be one of the options in the grace period dropdown selector.";
+    public static final String GRACE_PERIOD_NEGATIVE_ERROR_MESSAGE = "Grace period should not be negative," + " "
```
Comment (whipermr5) [inline src/main/java/teammates/common/util/FieldValidator.java]: Should end with a period not comma.

**Comment by shaharyar-shamshi** [comment]
@whipermr5 Sir suggested changes made kindly review it
